### DISCUSSION TITLE
Which roles to select for the 2nd account that's being invited?

### DISCUSSION THREAD

**Discussion by backslashV** [homeassistant-apps/app-cloudflared#550] [Q&A]
Super Administrator - All Privileges
Can edit any Cloudflare setting, make purchases, update billing, and manage memberships. Super Administrators can revoke the access of other Super Administrators.

Vectorize Readonly
Can read Vectorize configurations.

Vectorize Admin
Can edit Vectorize configurations.

Bot Management (Account-Wide)
Can edit Bot Management (including Super Bot Fight Mode) configurations for all domains in account

Hyperdrive Readonly
Can read Hyperdrive database configurations.

Hyperdrive Admin
Can edit Hyperdrive database configurations.

Turnstile Read
Grants read access to Turnstile

Turnstile
Grants full access to Turnstile

API Gateway Read
Grants read access to API Gateway (including API Shield) for all domains in an account

API Gateway
Grants full access to API Gateway (including API Shield) for all domains in an account

Cloudflare R2 Read
Can read R2 buckets, objects, and associated configurations.

Cloudflare R2 Admin
Can edit R2 buckets, objects, and associated configurations.

Page Shield Read
Grants read access to Page Shield across the whole account

Page Shield
Grants write access to Page Shield across the whole account

Cloudflare DEX
Can edit Cloudflare DEX.

Zone Versioning Read (Account-Wide)
Can view Zone Versioning for all domains in account

Zone Versioning (Account-Wide)
Can view and edit Zone Versioning for all domains in account'

Minimal Account Access
Can view account, and nothing else

Zaraz Edit
Can edit Zaraz configuration.

Waiting Room Admin
Can edit Waiting Room configuration.

Waiting Room Read
Can read waiting rooms configuration.

Magic Network Monitoring Admin
Can view, edit, create, and delete MNM configuration

Magic Network Monitoring
Can view and edit MNM configuration

Magic Network Monitoring Read-Only
Can view MNM configuration

Network Services Write (Magic)
Grants write access to network configurations for Magic services.

Network Services Read (Magic)
Grants read access to network configurations for Magic services.

HTTP Applications
Can view and edit HTTP Applications

HTTP Applications Read
Can view HTTP Applications

Trust and Safety
Can view and request reviews for blocks

Zaraz Admin
Can edit and publish Zaraz configuration.

Zaraz Readonly
Can read Zaraz configuration.

Cloudflare Images
Can edit Cloudflare Images assets

Cloudflare Gateway
Can edit Cloudflare Gateway and read Access.

Cloudflare Zero Trust Reporting
Can access Cloudflare for Zero Trust reporting data.

Cloudflare Zero Trust Read Only
Can access Cloudflare for Zero Trust read only mode.

Cloudflare Zero Trust
Can edit Cloudflare Zero Trust.

Cloudflare Zero Trust PII
Can access Cloudflare Zero Trust PII.

SSL/TLS, Caching, Performance, Page Rules, and Customization
Can edit most Cloudflare settings except for DNS and Firewall.

Log Share Reader
Can read Enterprise Log Share.

Log Share
Can edit Log Share configuration.

Load Balancer
Can edit Load Balancers, Pools, Origins, and Health Checks.

Firewall
Can edit WAF, IP Firewall, and Zone Lockdown settings.

DNS
Can edit DNS records.

Cloudflare Workers Admin
Can edit Cloudflare Workers.

Cloudflare Stream
Can edit Cloudflare Stream media.

Cloudflare Access
Can edit Cloudflare Access.

Cache Purge
Can purge the edge cache.

Billing
Can edit the account’s billing profile and subscriptions.

Audit Logs Viewer
Can view Audit Logs.

Analytics
Can read Analytics.

Administrator Read Only
Can access the full account in read-only mode.

Administrator
Can access the full account, except for membership management and billing.

**Comment by brenner-tobias** [comment]
Thanks for raising this. I am not sure myself to be honest, but my suggestion would be to work your way up, starting from very limited access.
I assume that you do need "DNS" (for creation of DNS entries of external_hostname to the tunnel) and "Cloudflare Zero Trust" (for the actual creation of the tunnel in the Zero Trust Dashboard).

Let me know what works for you

**Reply by backslashV** [reply]
Thank you for your response. I have enabled those two. However, when trying to set up the addon using my secondary account, I get this error "Unauthorized to access requested resource (Code: 9109)".

**Reply by brenner-tobias** [reply]
I suggest trying some others then. This is also something you can ask in the official [Cloudflare Community](https://community.cloudflare.com/), which is pretty active. It would be great if you could report back of link the discussion here.

**Reply by backslashV** [reply]
I think I tried over 20 combinations so far. There are so many of them that it's impossible to try every one. I've already asked in the community but haven't received a response in a week.

**Reply by brenner-tobias** [reply]
Did you figure out any combination that works or are you using Administrator?

**Reply by backslashV** [reply]
Nope. I'm using admin.
